File: CP/credit_suisse/GSUB.cpp
// good problem
// segment tree
//GSUB codechef

#include<bits/stdc++.h>
using namespace std;
#define endl "\n"
const int mxn = 2e5+2;

int a[mxn];
struct Node{
    int l,r,size;
}st[4*mxn];

Node build(int root, int l, int r){
    if(l==r){
        Node node ={a[l],a[l],1};
        st[root] = node;
        return st[root];
    }
    int mid = l+r>>1;
    Node ll = build(2*root+1, l, mid);
    Node rr = build(2*root+2, mid+1, r);
    if (ll.r==rr.l)
        ll.size--;
    return st[root] = {ll.l, rr.r, ll.size+rr.size};
}

Node update(int root, int l, int r, int idx){
    if (idx<l || idx>r) return st[root];
    if (l==r) {
        return st[root] = {a[l],a[l],1};
    }
    int mid = l+r>>1;
    Node ll,rr;
    if (idx<=mid){
        ll = update(2*root+1, l, mid, idx);
        rr = st[2*root+2];
    }
    else{
        rr = update(2*root+2, mid+1, r, idx);
        ll = st[2*root+1];
    }
    if (ll.r==rr.l){
        ll.size--;
    }
    return st[root] = {ll.l, rr.r, ll.size+rr.size};
}

/*int rec(int l, int r){
    if (l==r) return 1;
    int mid = l+r>>1;
    int res = rec(l,mid)+rec(mid+1,r);
    if (a[mid]==a[mid+1])
        res--;
    return res;
}*/
int main(){
    // freopen("in.txt", "r", stdin);
    ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    int t = 1;
    cin>>t;
    while(t-->0){
        int n,q;
        cin>>n>>q;
        for(int i=0; i<n; ++i) cin>>a[i];
        build(0, 0, n-1);
        int x,y;
        for (int i =0; i<q; ++i){
            cin>>x>>y;
            x--;
            a[x] = y;
            cout<<update(0,0,n-1,x).size<<endl;
            // cout<<rec(0,n-1)<<endl;
        }
    }
    return 0;
}


//
 // freopen("in.txt", "r", stdin);

File: CP/credit_suisse/GSUB.py
import os, sys, bisect
from collections import defaultdict, Counter, deque;
from functools import lru_cache   #use @lru_cache(None)
if os.path.exists('in.txt'): sys.stdin=open('in.txt','r')
if os.path.exists('out.txt'): sys.stdout=open('out.txt', 'w')
#
input = lambda: sys.stdin.readline().strip()
imap = lambda: map(int,input().split()); ilist = lambda: list(imap())
#------------------------------------------------------------------


import sys
sys.setrecursionlimit(10**7)

input = sys.stdin.readline
for _ in range(int(input())):
    n,q = map(int,input().split())
    a = list(map(int,input().split()))
    # def rec(l,r):
    #     if l==r:
    #         return 1
    #     mid = (l+r)>>1
    #     res = rec(l,mid)+rec(mid+1,r)
    #     if a[mid]==a[mid+1]:
    #         res-=1
    #     # print(l,r,"-",res,"'",a[mid],a[mid+1])
    #     return res

    st = [[] for i in range(4*n+1)]
    def build(root, l,r):
        if l==r:
            st[root] = [a[l],a[l],1]
            return [a[l],a[l],1]
        mid = (l+r)>>1
        ll = build(2*root+1, l,mid)
        rr = build(2*root+2, mid+1,r)
        if ll[1]==rr[0]:
            ll[2]-=1
        st[root] = [ll[0],rr[1], ll[2]+rr[2]]
        return [ll[0],rr[1], ll[2]+rr[2]]
    def update(root, l,r,idx):
        if idx<l or idx>r:
            return st[root]
        if l==r:
            st[root] = [a[l],a[l],1]
            return [a[l],a[l],1]
        mid = l+r>>1
        if idx<=mid:
            update(2*root+1, l,mid,idx)
            ll = st[2*root+1][:]
            rr = st[2*root+2][:]
            if ll[1]==rr[0]:
                ll[2]-=1
        else:
            update(2*root+2, mid+1,r,idx)
            ll = st[2*root+1][:]
            rr = st[2*root+2][:]
            if ll[1]==rr[0]:
                ll[2]-=1
        st[root] = [ll[0],rr[1], ll[2]+rr[2]]
        return st[root]
    build(0, 0,n-1)
    # print(st)
    for j in range(q):
        x,y = map(int,input().split())
        x-=1
        a[x] = y
        print(update(0, 0,n-1,x)[2])
        # print(st)

    # for j in range(q):
    #     x,y = map(int,input().split())
    #     a[x] = y
    #     res = 0
    #     cnt = 1
    #     b = []
    #     for i in range(1,n+1):
    #         if b and b[-1]==a[i]:
    #             continue
    #         b.append(a[i])
    #     # print(a)
    #     # res =max(cnt,res)
    #     print(len(b))


